tpm2/tpm2.go

```diff
@@ -906,24 +906,36 @@ func encodeQuote(signingHandle tpmutil.Handle, parentPassword, ownerPassword str
 	return concat(ha, auth, params, pcrs)
 }
 
-func decodeQuote(in []byte) ([]byte, *Signature, error) {
+func decodeQuote(in []byte) ([]byte, []byte, error) {
 	buf := bytes.NewBuffer(in)
 	var paramSize uint32
 	var attest tpmutil.U16Bytes
 	if err := tpmutil.UnpackBuf(buf, &paramSize, &attest); err != nil {
 		return nil, nil, err
 	}
-	sig, err := DecodeSignature(buf)
-	return attest, sig, err
+	rawSig := buf.Next(int(paramSize) - len(attest))
```

Comment:
I think this might have an "off by 2 error" as the attest structure is a TPM2B_ATTEST which has a two byte size before the buffer, so this would give the wrong lengths.

The code we want is something like:

```go
buf := bytes.NewBuffer(in)
var paramSize uint32
if err := tpmutil.UnpackBuf(buf, &paramSize); err != nil {
	return nil, nil, err
}
buf.Truncate(paramSize)
var attest tpmutil.U16Bytes
if err := tpmutil.UnpackBuf(buf, &paramSize, &attest); err != nil {
	return nil, nil, err
}
return attest, buf.Bytes(), nil
```

Reply:
Make sense to me.
attest doesn't have a paramSize in front of it, so I removed the paramSize